src/mono/mono/component/debugger-engine.c

```diff
@@ -89,7 +83,8 @@ domains_cleanup (void)
 void
 mono_de_foreach_domain (GHFunc func, gpointer user_data)
 {
-	g_hash_table_foreach (domains, func, user_data);
+	MonoDomain *domain = mono_domain_get ();
```

Comment:
Semantics:
* `mono_domain_get` - returns the domain of the current thread.  Either the root domain or `NULL` if the thread is not currently attached to the runtime.  (This is a TLS lookup)
* `mono_get_root_domain` - returns the root domain.  This is always non-NULL (except early during runtime startup) and is a global variable lookup.

`mono_de_foreach_domain` should probably just use `mono_get_root_domain`.  Although it's a bit academic because the debugger engine thread is always attached.  But maybe it's better to avoid a TLS lookup

Reply:
Thanks, I was thinking they were equivalent, but I agree we should probably use the root domain getter